recipes/historyserver.rb

```diff
@@ -0,0 +1,87 @@
+my_ip = my_private_ip()
+my_public_ip = my_public_ip()
+
+
+completed_jobs_dir = "#{node['flink']['historyserver']['remote_dir']}"
+
+hops_hdfs_directory completed_jobs_dir do
+    action :create_as_superuser
+    owner node['flink']['user']
+    group node['flink']['group']
+    mode "1733"
+end
+
+
+deps = ""
+if exists_local("hops", "nn")
+  deps = "namenode.service"
+end
+service_name="flinkhistoryserver"
+
+service service_name do
+    provider Chef::Provider::Service::Systemd
+    supports :restart => true, :stop => true, :start => true, :status => true
+    action :nothing
+  end
+
+case node['platform_family']
+when "rhel"
+  systemd_script = "/usr/lib/systemd/system/#{service_name}.service"
+else
+  systemd_script = "/lib/systemd/system/#{service_name}.service"
+end
+
+ruby_block "hadoop_glob" do
+  block do
+    Chef::Resource::RubyBlock.send(:include, Chef::Mixin::ShellOut)
+    node.default['hadoop_glob'] = shell_out("#{node['hops']['bin_dir']}/hadoop classpath --glob  | tr -d '\n'").stdout
+  end
+  action :create
+end
+
+template node['flink']['historyserver']['environment'] do
+    source "historyserver.env.erb"
+    owner node['flink']['user']
+    group node['flink']['group']
+    mode 0750
+    variables({
+      :hadoop_glob=> lazy { node['hadoop_glob'] },
+      :flink_log_dir=> node['flink']['historyserver']['logs']
+    })
+end
+
+template systemd_script do
+    source "#{service_name}.service.erb"
+    owner "root"
+    group "root"
+    mode 0754
+    variables({
+      :deps=> deps
+    })
+    notifies :enable, resources(:service => service_name)
+    notifies :start, resources(:service => service_name), :immediately
+  end
+
+kagent_config service_name do
+  action :systemd_reload
+end
+
+if node['services']['enabled'] != "true" && node['flink']['systemd'] == "true"
```

Comment:
I don't think you need to check for `node['flink']['systemd']` the rest of the recipe and the entire platform relies on Systemd.

Reply:
Can we remove this check for other cookbooks then?